src/test/java/address/testutil/ApplicationTest.java

```diff
@@ -0,0 +1,80 @@
+package address.testutil;
+
+import guitests.GuiRobot;
+import javafx.application.Application;
+import javafx.application.HostServices;
+import javafx.application.Preloader;
+import javafx.stage.Stage;
+import org.junit.After;
+import org.junit.Before;
+import org.testfx.api.FxToolkit;
+import org.testfx.api.annotation.Unstable;
+import org.testfx.framework.junit.ApplicationAdapter;
+import org.testfx.framework.junit.ApplicationFixture;
+
+/**
+ * Code mostly from TestFX
+ */
+public abstract class ApplicationTest extends GuiRobot implements ApplicationFixture {
+
+    //---------------------------------------------------------------------------------------------
+    // STATIC METHODS.
+    //---------------------------------------------------------------------------------------------
+
+    @Unstable(reason = "is missing apidocs")
+    public static void launch(Class<? extends Application> appClass,
+                              String... appArgs) throws Exception {
+        FxToolkit.registerPrimaryStage();
+        FxToolkit.setupApplication(appClass, appArgs);
+    }
+
+    //---------------------------------------------------------------------------------------------
+    // METHODS.
+    //---------------------------------------------------------------------------------------------
+
+    @Before
+    @Unstable(reason = "is missing apidocs")
+    public final void internalBefore()
+            throws Exception {
+        FxToolkit.registerPrimaryStage();
+        FxToolkit.setupApplication(() -> new ApplicationAdapter(this));
+        sleep(1000);
```

Comment:
Is this class simply to add the sleep duration? I think we can extend `ApplicationTest` instead?